ironfish-rust-nodejs/src/frost.rs

```diff
@@ -237,3 +237,34 @@ pub fn split_secret(
         public_key_package: bytes_to_hex(&public_key_package_vec),
     })
 }
+
+#[napi(js_name = "PublicKeyPackage")]
+pub struct NativePublicKeyPackage {
+    public_key_package: PublicKeyPackage,
+}
+
+#[napi]
+impl NativePublicKeyPackage {
+    #[napi(constructor)]
+    pub fn new(value: String) -> Result<NativePublicKeyPackage> {
+        let bytes = hex_to_vec_bytes(&value).map_err(to_napi_err)?;
+
+        let public_key_package = PublicKeyPackage::read(&bytes[..]).map_err(to_napi_err)?;
+
+        Ok(NativePublicKeyPackage { public_key_package })
+    }
+
+    #[napi]
+    pub fn identities(&self) -> Result<Vec<Buffer>> {
+        let mut identities: Vec<Buffer> = vec![];
+
+        for identity in &self.public_key_package.identities {
+            let mut vec: Vec<u8> = vec![];
+            identity.serialize_into(&mut vec).map_err(to_napi_err)?;
+
+            identities.push(Buffer::from(vec))
+        }
+
+        Ok(identities)
```

Comment:
I believe this can be rewritten in a way that is infallible and reduces the number of allocations (untested):

```rust
self.public_key_packages.identities.iter().map(|id| Buffer:from(&identity.serialize()[..])).collect()
```